File: bridge/YouTubeVideoProvider.java
package bridge;

public class YouTubeVideoProvider extends VideoProvider {

    private final VideoQuality videoQuality;

    public YouTubeVideoProvider(VideoQuality videoQuality) {
        this.videoQuality = videoQuality;
    }

    @Override
    public void playback(String videoId) {
        videoQuality.render();
        System.out.println("Playback youtube video...");
    }
}

File: typescript/patterns/bridge/youtube-video-provider.ts
class YoutubeVideoProvider extends VideoProvider {
  private videoQuality: VideoQuality;

  constructor(videoQuality: VideoQuality) {
    super();
    this.videoQuality = videoQuality;
  }
  override playback(videoId: string) {
    this.videoQuality.render();
    console.log("Playback youtube video...");
  }
}
